common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -154,16 +154,24 @@ public static class Collation {
      */
     public final boolean supportsLowercaseEquality;
 
+    /**
+     * Support for Space Trimming implies that that based on specifier (for now only right trim)
+     * leading, trailing or both spaces are removed from the input string before comparison.
+     */
+    public final boolean supportsSpaceTrimming;
+
     public Collation(
         String collationName,
         String provider,
         Collator collator,
         Comparator<UTF8String> comparator,
         String version,
         ToLongFunction<UTF8String> hashFunction,
+        BiFunction<UTF8String, UTF8String, Boolean> equalsFunction,
         boolean supportsBinaryEquality,
         boolean supportsBinaryOrdering,
-        boolean supportsLowercaseEquality) {
+        boolean supportsLowercaseEquality,
+        boolean supportsSpaceTrimming) {
```

Comment:
Is it possible to have binaryEquality and spaceTrimming at the same time? If not could we add another assert?

Reply:
this is a good point, for example if we have "UTF8_BINARY_RTRIM", then it's no longer "binaryEquality" per se

however, I suggest we visit this issue in a follow-up, additional assert could be good

Reply:
Yes, actually i thought about this and plan to address it in follow-up pr.
So yes, this name is not accurate any more. One could say, why just not then set it on false for UTF8_BINARY collations using trimming? Well for many expressions trim specifier is passthrough and we would like to have it done together with UTF8_BINARY collation. Not to mention on many places supportsBinaryEquality = isUtfBinary which is ok i guess because I guess it was not planeed to extend UTF8_binary collaiton.
So simple solution for this would be to **rename supportsBinaryOrdering to something like "isUtf8BinaryType"**.

Reply:
I created Jira task for this:
https://issues.apache.org/jira/browse/SPARK-49911